eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/ManagementInputLibrary.cs

```diff
@@ -10,14 +13,127 @@ namespace Azure.Generator.Management
+            var result = new Dictionary<string, InputModelType>();
+            foreach (var model in InputNamespace.Models)
+            {
+                if (!result.ContainsKey(model.CrossLanguageDefinitionId))
+                {
+                    result.Add(model.CrossLanguageDefinitionId, model);
+                }
+            }
+            return result;
+        }
+
+        private IReadOnlyDictionary<InputClient, ResourceMetadata> BuildResourceMetadata()
+        {
+            var resourceMetadata = new Dictionary<InputClient, ResourceMetadata>();
+            foreach (var client in AllClients)
+            {
+                var decorator = client.Decorators.FirstOrDefault(d => d.Name == KnownDecorators.ResourceMetadata);
+                if (decorator != null)
+                {
+                    var metadata = BuildResourceMetadata(decorator);
+                    resourceMetadata.Add(client, metadata);
+                }
+            }
+            return resourceMetadata;
+
+            ResourceMetadata BuildResourceMetadata(InputDecoratorInfo decorator)
+            {
+                var args = decorator.Arguments ?? throw new InvalidOperationException();
+                string? resourceType = null;
+                InputModelType? resourceModel = null;
+                InputClient? resourceClient = null;
+                bool isSingleton = false;
+                ResourceScope resourceScope = default;
+                if (args.TryGetValue(KnownDecorators.ResourceType, out var resourceTypeData))
+                {
+                    resourceType = resourceTypeData.ToObjectFromJson<string>();
+                }
+
+                if (args.TryGetValue(KnownDecorators.ResourceModel, out var resourceModelData))
+                {
+                    var resourceModelId = resourceModelData.ToObjectFromJson<string>();
+                    if (resourceModelId != null)
+                    {
+                        resourceModel = GetModelByCrossLanguageDefinitionId(resourceModelId!);
+                    }
+                }
+
+                if (args.TryGetValue(KnownDecorators.ResourceClient, out var resourceClientData))
+                {
+                    var resourceClientId = resourceClientData.ToObjectFromJson<string>();
+                    if (resourceClientId != null)
+                    {
+                        resourceClient = GetClientByCrossLanguageDefinitionId(resourceClientId!);
+                    }
+                }
+
+                if (args.TryGetValue(KnownDecorators.IsSingleton, out var isSingletonData))
+                {
+                    isSingleton = isSingletonData.ToObjectFromJson<bool>();
+                }
+
+                if (args.TryGetValue(KnownDecorators.ResourceScope, out var scopeData))
+                {
+                    var scopeString = scopeData.ToObjectFromJson<string>();
+                    resourceScope = Enum.Parse<ResourceScope>(scopeString!, true);
```

Comment:
what if the scopeString in not defined in ResourceScope?

Reply:
my idea is that all these things should be required in order to get a correct resource.
My code in the emitter should ensure this

Reply:
sure, but the best practice of parsing enum should be like
https://github.com/Azure/azure-sdk-for-net/blob/af7c6e192b5b2aa880053424e7bf95e44a9b231c/sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs#L125-L132
And there could be potentially mistakes from the emitter side

Reply:
I would not say this is the "best practice".
I think if there is a best practice of parsing enums, we should use `TryParse` instead of checking if it is defined first and then parsing it.
I could change to TryParse, but I think I have written some comments here:
I am throwing when any of these things are not properly parsed.
I know we should not throw exceptions, but I did that here and in the next PR or in the future we could discuss how to deal with it.
It is very possible that the emitter could be wrong or not correctly implemented to produce errors in the tspCodeModel, and how we handle that?

Now, because we are in a very early stage, my choice is to crash everything so that we could see there is a problem. I do not want to make everything built on a "bug".